src/v/cluster/cloud_metadata/cluster_manifest.h

```diff
@@ -37,6 +34,15 @@ struct cluster_metadata_manifest
   , public cloud_storage::base_manifest {
     using duration = std::chrono::milliseconds;
 
+    auto serde_fields() {
+        return std::tie(
+          upload_time_since_epoch,
+          cluster_uuid,
+          metadata_id,
+          controller_snapshot_offset,
+          controller_snapshot_path);
+    }
```

Comment:
presumably serde wouldn't have even worked before (despite being an envelope) with the automagic reflection since it had a base class which i'd assume would disable that feature?

Reply:
Right, I was lazy earlier because I didn't need serde at the time (when uploading, this manifest is serialized as JSON, but in an upcoming PR it will be serde-ed into the controller log)